src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -416,56 +418,101 @@ def initialize_new_simulants_from_acs(
 
         return new_simulants
 
-    def initialize_simulant_link_columns(self, pop: pd.DataFrame) -> pd.DataFrame:
+    def initialize_simulant_link_columns(
+        self, new_simulants: pd.DataFrame, existing_simulants: Union[pd.DataFrame, None]
+    ) -> pd.DataFrame:
         """
         Initializes columns that link simulants to one another.
         These are name (currently, only last name actually links) and guardian_1 and guardian_2.
         This depends on all simulants being initialized and having their final indices.
 
+        Note that new simulants may be linked to other new simulants.
+
         Parameters
         ----------
-        pop
-            DataFrame of simulants, containing at least household_id, relation_to_household_head, age, and race_ethnicity,
+        new_simulants
+            DataFrame of new simulants who should be assigned values for these linking columns,
+            containing at least household_id, relation_to_household_head, age, and race_ethnicity,
```

Comment:
Hm yeah -- what I've listed here is exactly what is needed, but I suppose I could just say that all columns besides the linking columns should be initialized already. Because of `columns_needed_for_linking` below (a) the specific list of columns needed isn't hard to figure out by looking at the method and (b) it will fail in a pretty easy-to-understand way if these columns don't exist.